packages/token-balance-cache/src/tbc.ts

```diff
@@ -224,7 +224,11 @@ export class TokenBalanceCache
     // grab contract if provided, otherwise query native token
     let opts = {};
     if (contractAddress) {
-      if (network !== ChainNetwork.ERC20 && network !== ChainNetwork.ERC721) {
+      if (
+        network !== ChainNetwork.ERC20 &&
+        network !== ChainNetwork.ERC721 &&
+        network !== ChainNetwork.ERC1155
```

Comment:
ERC1155 doesn't work for this legacy method `fetchUserBalance()` because it doesn't have provided the `tokenId`! We can just remove that here, because it'll work for the general forward-proofed method `getBalancesForAddresses()`. Dig into the logic here and see if that makes sense to you too!

Reply:
This logic makes much more sense now. I see that fetchUserBalance has fixed opts, so that tokenId is not included whereas getBalancesForAddresses() take in a opts type that would allow for tokenId!